hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -1371,80 +1115,35 @@ def _draw_bounded_integer(
-            # zero - if that happens then often the data stream becomes misaligned
-            # and we fail to shrink in cases where we really should be able to.
-            self._cd.draw_bits(1, forced=0)
-            return int(lower)
-
-        if center is None:
-            center = lower
-        center = min(max(center, lower), upper)
-
-        if center == upper:
-            above = False
-        elif center == lower:
-            above = True
-        else:
-            force_above = None if forced is None else forced < center
-            above = not self._cd.draw_bits(
-                1, forced=force_above, fake_forced=fake_forced
-            )
+        assert self._cd._random is not None
 
-        if above:
-            gap = upper - center
-        else:
-            gap = center - lower
-
-        assert gap > 0
-
-        bits = gap.bit_length()
-        probe = gap + 1
+        if lower == upper:
+            return lower
 
-        if (
-            bits > 24
-            and _vary_effective_size
-            and self.draw_boolean(
-                7 / 8, forced=None if forced is None else False, fake_forced=fake_forced
-            )
-        ):
-            # For large ranges, we combine the uniform random distribution from draw_bits
+        bits = (upper - lower).bit_length()
+        if bits > 24 and vary_size and self._cd._random.random() < 7 / 8:
+            # For large ranges, we combine the uniform random distribution
             # with a weighting scheme with moderate chance.  Cutoff at 2 ** 24 so that our
             # choice of unicode characters is uniform but the 32bit distribution is not.
             idx = INT_SIZES_SAMPLER.sample(self._cd)
-            force_bits = min(bits, INT_SIZES[idx])
-            forced = self._draw_bounded_integer(
-                lower=center if above else max(lower, center - 2**force_bits - 1),
-                upper=center if not above else min(upper, center + 2**force_bits - 1),
-                _vary_effective_size=False,
-            )
-
-            assert lower <= forced <= upper
-
-        while probe > gap:
-            probe = self._cd.draw_bits(
-                bits,
-                forced=None if forced is None else abs(forced - center),
-                fake_forced=fake_forced,
+            cap_bits = min(bits, INT_SIZES[idx])
+            result = self._draw_bounded_integer(
+                lower=lower,
+                upper=min(upper, lower + 2**cap_bits - 1),
+                vary_size=False,
             )
+            assert lower <= result <= upper
+            return result
 
-        if above:
-            result = center + probe
-        else:
-            result = center - probe
+        result = upper + 1
+        while result > upper:
+            result = lower + self._cd._random.getrandbits(bits)
 
         assert lower <= result <= upper
-        assert forced is None or result == forced, (result, forced, center, above)
         return result
```

Comment:
I think we can just `return self._cd._random.randint(lower, upper)` here?  And similarly in the smaller-interval case above; it should be quicker to do any retries in C rather than Python.

Reply:
I think `randint` is in python unless vscode is redirecting me to a fallback-only c implementation - but thanks for pointing this out, this is better regardless.